packages/server/graphql/types/User.ts

```diff
@@ -554,21 +554,41 @@ const User: GraphQLObjectType<any, GQLContext> = new GraphQLObjectType<any, GQLC
     },
     teams: {
       type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(Team))),
-      description: 'all the teams the user is on that the viewer can see.',
+      description: 'all the teams visible to the user',
       resolve: async (
         {id: userId}: {id: string},
         _args: unknown,
         {authToken, dataLoader}: GQLContext
       ) => {
         const viewerId = getUserId(authToken)
-        const user = (await dataLoader.get('users').load(userId))!
-        const teamIds =
-          viewerId === userId || isSuperUser(authToken)
-            ? user.tms
-            : user.tms.filter((teamId: string) => authToken.tms.includes(teamId))
-        const teams = (await dataLoader.get('teams').loadMany(teamIds)).filter(isValid)
-        teams.sort((a, b) => (a.name > b.name ? 1 : -1))
-        return teams
+        const [user, organizationUsers] = await Promise.all([
+          dataLoader.get('users').load(userId),
+          dataLoader.get('organizationUsersByUserId').load(userId)
+        ])
+        const orgIds = organizationUsers.map(({orgId}) => orgId)
+
+        const organizationTmsWithPublicFlag = []
+
+        for (const orgId of orgIds) {
+          const organization = await dataLoader.get('organizations').load(orgId)
+          if (organization.featureFlags?.includes('publicTeams')) {
```

Comment:
-2 I did not test this, but for permissions we previously checked that the `viewer` (the one making the request) can only see the teams they're already on.
Right now it seems that I could see the teams I'm already on plus all the teams from organizations with the public feature flag, regardless of whether or not I'm on that org.

Reply:
I couldn't let it go and tried again and could reproduce the error I was seeing in the code: https://www.loom.com/share/e3fb1b29958d4da08166a3f965a17759